src/Mirza/SupplyChain/EventUtils.hs

```diff
@@ -189,31 +207,26 @@ findClassLabelId' cp msfv ir lot = do
              Schema.label_item_reference labels ==. (val_ . Just $ ir)
            )
     pure labels
-  pure $ Schema.label_id <$> l
+  pure $ case l of
+    [lbl] -> Just $ Schema.label_id lbl
```

Comment:
I find this pattern a little concerning, does beam offer a _select one or none_ helper?

If not, can we pattern match on lists length > 1 and fail explicitly? Maybe a helper function

Reply:
As you correctly identified, `runSelectReturningOne` from beam is the helper function that does a `select one or none`. I suppose the right thing to do here in case we get a list with length > 1 is to construct an error type.

Reply:
I'd even settle for an `error "unexpected row length"` to begin with, anything to fail fast if we ever get  more than 1 result back.